go/ethadapter/blockprovider_test.go

```diff
@@ -171,6 +171,11 @@ func (e *ethClientMock) BlockByNumber(num *big.Int) (*types.Block, error) {
 	return block, nil
 }
 
+func (e *ethClientMock) BlockNumber() (uint64, error) {
```

Comment:
This is a new method we added to L1 clients, so we have to implement it on the mock clients now.